### ISSUE TITLE
Specific indentations

### ISSUE THREAD

**Issue by sbcgua** [larshp/abaplint#337]
arguable, just an opinion

case 1, try-catch.
catch should not be indented
```abap
    try.
      call_a_method( ).
    catch cx_static_check into lx. " <<<<<
      process_error( lx ).
    endtry.
```

case 2, indentation right after class definition/implementation
when created with se80 2nd level code is not indented.
no problem to indent the definition. But indenting the implementation will add starting spaces when editing a specific method - which looks ugly imho.
```abap
class ZCL_MY_CLASS definition
  public
  final
  create private .

public section. " <<<<<
  ...
class ZCL_MY_CLASS implementation.
method CONSTRUCTOR. " <<<<<
  data ...
```

I know the above is the matter of taste of course. Could it be a good idea to add some params to indentation rule to exclude specific cases like above? (one by one of course). E.g:

```json
    "indentation":{  
      "enabled": true,
      "try-catch-indent": false,
      "class-def-indent": false,
    }
```

**Comment by larshp** [comment]
yeah, we can add it in the config, its not much work to add in abaplint

currently the rules are built to follow the SAP standard pretty printer, which also follows the help.sap.com examples, https://help.sap.com/doc/abapdocu_751_index_htm/7.51/en-US/abaptry.htm https://help.sap.com/doc/abapdocu_750_index_htm/7.50/en-US/abapclass_definition.htm

out of curiosity, does this mean you manually indent everything without using the pretty printer?

**Comment by sbcgua** [comment]
if you hint where to look and an examples, i can try to PR if you don't mind :)

pretty printer: yes, almost never :) Why? Mmmm: don't know ... historically so, maybe I don't trust them as they sometimes ruin non-standartly formatted but well readable parts of code ... :)

**Comment by larshp** [comment]
the indentation rule starts in https://github.com/larshp/abaplint/blob/master/src/rules/indentation.ts

will need to add more options to `IndentationConf `

and then pass to https://github.com/larshp/abaplint/blob/master/src/abap/pretty_printer.ts

**Comment by sbcgua** [comment]
seems like current pretty printer has own issues. E.g 
```js
getExpectedIndentation("try.\nwrite moo.\ncatch.\nwrite err.\nendtry.") = [1,5,5,5,1] // 5 for catch
```
Can it be ?
BTW is it possible to output pretty printed result ? I mean is there a command line param for that ?

**Comment by sbcgua** [comment]
regarding the case 2 (with global class indentation): how do I find that the class is global ? (in the context of `indentation.ts`)

Is it something like `statement.getChildren().find(t => t instanceof Statements.ClassGlobal)` ?

P.S. BTW why do you enforce `ter-arrow-body-style always` in tslint ? The above statement is incorrect but it far better readable apart from `...find(t => { return t instanceof Statements.ClassGlobal)}`

**Comment by larshp** [comment]
"makes the above statement incorrect", how so?

**Comment by sbcgua** [comment]
OK, the second rule is also ready. So will PR right after the first one merged.

```
statement.getChildren().find(t => t instanceof Statements.ClassGlobal)
>>> tslint: Expected block statement surrounding arrow body. (ter-arrow-body-style)
```
But anyway it is more to personal preferences

**Comment by larshp** [comment]
add a local flag variable? and save the information, the global defintion and global implementation is always in the same file

**Comment by sbcgua** [comment]
I though there is some index of found tokens available. But it is an option too. Will PR soon.

**Comment by sbcgua** [comment]
I think can be closed now

**Comment by DHEWC001** [comment]
Hi Guys,
I'm new to github and trying to write some abap code there. I'm getting Indentation error while defining a PUBLIC SECTION in the local class.
REPORT zdemo.

CLASS lcl_class DEFINITION.

PUBLIC SECTION.   <<< This line of code is giving Indentation error. Any clue??

ENDCLASS.

CLASS lcl_class IMPLEMENTATION.

ENDCLASS.
![image](https://user-images.githubusercontent.com/21343826/55602308-e0064780-5732-11e9-8867-d39dec2ea434.png)

Please help.

**Comment by sbcgua** [comment]
Hi @DHEWC001,
`public section` kind of belong to class so it should be 

```abap
class lcl_class definition.
  public section. <<< indent it too !
    methods xxx.
    data yyyy type i.
endclass.
```

However in the context of this particular issue you can also specify `globalClassSkipFirst = true` in `indentation` rule of the abaplint.json to suppress particulary first class line indentation.

```abap
class ZCL_MY_CLASS implementation.
method CONSTRUCTOR. " <<<<<
  data ...
```
